appinventor/components/src/com/google/appinventor/components/runtime/TextBoxBase.java

```diff
@@ -499,6 +539,30 @@ public void TextColor(int argb) {
     }
   }
 
+  @SimpleFunction(description = "Repositions the cursor of the TextBox before the character at the given" +
+   " 1-indexed position. If the given position is larger than the length of the `TextBox`, the cursor will be moved" +
+   " to the end of the text; and if the given position is smaller or equal to 1, the cursor will be moved to the front.")
+  public void MoveCursorTo(int position) {
+    int len = view.getText().toString().length();
+    if (position > len) {
+      view.setSelection(len);
+    } else if (position <= 1) {
+      view.setSelection(0);
+    } else {
+      view.setSelection(position - 1);
+    }
+  }
+
+  @SimpleFunction(description = "Repositions the cursor to the end of the %type%.")
+  public void MoveCursorToEnd() {
+    MoveCursorTo(view.getText().length());
+  }
+
+  @SimpleFunction(description = "Repositions the cursor to the front of the %type%.")
+  public void MoveCursorToFront() {
```

Comment:
I think I would prefer either Start or Beginning rather than Front to complement the use of End. For example, a story has a beginning, middle, and end, but it would seem weird to say let's go to the front of the story.

Reply:
Updated.